rock/python.rb

```diff
@@ -0,0 +1,50 @@
+require 'open3'
+
+module Rock
+    # Identify the available python version, per default checks
+    # Autobuild.programs['python'] setting
+    # @param[String] python_bin Path to a custom python version
+    def self.find_python_bin(python_bin = nil)
+        if !python_bin
+            python_bin = Autobuild.programs['python']
+            if !python_bin
+                python_bin = `which python`.strip()
+                if $? != 0
+                    python_bin = ""
+                end
+            end
+        end
+        if File.exist?(python_bin)
+            python_bin
+        else
+            nil
+        end
+    end
+
+    # Allow to update the PYTHONPATH for package, tries to guess the python
+    # binary from Autobuild.programs['python'] and system's default setting
+    # @param[Autobuild::Package] pkg
+    # @param[String] python_bin Path to a custom python version
+    def self.activate_python(pkg, python_bin = nil)
+        python_bin = Rock.find_python_bin(python_bin)
+        if python_bin
+            cmd = "#{python_bin} -c \"import sys;"\
```

Comment:
Same here, best would be to cache the version. This is most probably rather expensive to run, and all the blocks are run at each autoproj load.